src/pymodule.rs

```diff
@@ -129,3 +129,62 @@ fn get_input_algorithm(algorithm: &String) -> Result<BlendAlgorithm, PyErr> {
         )))),
     }
 }
+
+fn validate_algorithms(
```

Comment:
is this the proper name for this function... it basically converts a sequence of algorithms to a vector of algorithms and parameters right ?